recirq/otoc/otoc_circuits.py

```diff
@@ -0,0 +1,571 @@
+            contained in the corresponding element of the list.
+        reverse_ops: The two-qubit gates assigned to each qubit pair in
+            $U^\dagger$, having the same format as forward_ops.
+        butterfly_qubits: The qubits to which a perturbation between $U$ and
+            $U^\dagger$ is to be applied.
+        cycles_per_echo: How often a spin-echo pulse (a Y gate) is to be
+            applied to the ancilla during its idling time. For example,
+            if specified to be 2, a Y gate is applied every two cycles.
+        random_seed: The seed for the random single-qubit gates. If
+            unspecified, no seed will be used.
+        sq_gates: Indices for the random single-qubit gates. Dimension should
+            be len(qubits) $\times$ cycle. The values should be integers from 0
+            to 7 if z_only is False, and floats between -1 and 1 if z_only is
+            True.
+        cz_correction: Correction to the composite CZ gate, obtained from
+            parallel-XEB. If unspecified, the ideal decomposition for CZ into
+            sqrt-iSWAP and single-qubit gates will be used.
+        use_physical_cz: If True, a direct CZ gate will be used (instead of
+            composite gate from sqrt-iSWAP and single-qubit gates).
+        light_cone_filter: If True, gates outside the light-cones of the
+            butterfly qubit(s) will be removed and replaced with spin-echo
+            gates. Gates outside the light-cone of the measurement qubit (
+            i.e. qubits[0]) will be completely removed.
+        cliffords: If True (and sq_gates is unspecified), Clifford gates
+            drawn randomly from +/-X/2 and +/-Y/2 will be used for the
+            single-qubit gates.
+        z_only: If True, only Z gates (with a random exponent between -1 and
+            1) will be used for the random single-qubit gates.
+        padding: If specified, a delay with the given value (in nano-seconds)
+            will be added between $U$ and $U^\dagger$.
+        layer_by_layer_cal: If True, the two-qubit gates in forward_ops and
+            reverse_ops are different in each layer (i.e. forward_ops and
+            reverse_ops are specified as List[_CORRECTION_FILE]).
+        idle_echo_frequency: How often spin-echo pulses (random +/-X or +/-Y
+            gates) are applied to qubits outside the light-cones of the
+            butterfly qubit(s). Has no effect if light_cone_filter is False.
+        idle_echo_seed: The seed for the random spin-echo gates for qubits
+            outside the light-cones of the butterfly qubit(s). If
+            unspecified, no seed will be used. Has no effect if
+            light_cone_filter is False.
+
+    Returns:
+        A list of 16 cirq.Circuits. The first 4 circuits are OTOC circuits
+        with the butterfly operators being I (i.e. no perturbation between U
+        and $U^\dagger$). Circuits 4 to 8, 8 to 12 and 12 to 16 correspond to
```

Comment:
There's a ton of structured coded into the indices in this one flat list. Consider creating a small, self-documenting dataclass to clear this up. I think I made a note when I was confused about this reading through the example

```python
@dataclass
OTOCCircuits:
  butterfly_I: List[cirq.Circuits]
  butterfly_X: List[
  butterfly_Y ...
```

Reply:
Good suggestion! Done.